.github/workflows/codeql-analysis.yml

```diff
@@ -11,6 +11,11 @@
 #
 name: "CodeQL"
 
+# only one can run at a time.
```

Comment:
This looks like it would cause PRs with the same head_ref to cancel each other right?

Reply:
Right a meant that 2 or more PRs with the same head branch will cancel each other. Is there a way to identify individual PRs?

Reply:
umm head_ref is the source branch for the PR, would it be separate for separate PRs?

![image](https://user-images.githubusercontent.com/76567250/223525091-6868d117-2bfb-4461-87a8-1e4b9c526402.png)

Reply:
What about using `github.event.number`? Pull request events are:
- on open: `open` https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=opened#pull_request
- on push: `synchronize` https://docs.github.com/webhooks-and-events/webhooks/webhook-events-and-payloads?actionType=synchronize#pull_request

Reply:
but event number will not make the same PR commits in the same group for the new ones to cancel the last running